engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/DockerService.java

```diff
@@ -0,0 +1,166 @@
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+
+@Slf4j
+public class DockerService extends RService implements AutoCloseable {
+    private static final String WORKDIR = "/etc/analysis";
+    private final DockerClient client;
+    /**
+     * Directory prefix on HOST where the analysis files are unpacked.
+     */
+    @Value("${analysis.mount}")
+    private String analysisMount;
+    /**
+     * Socket to connect
+     */
+    @Value("${docker.host:#{null}}")
+    private String host;
+
+    @Value("${docker.image.default}")
+    private String defaultImage;
+
+    @Autowired
+    @Qualifier("analysisTaskExecutor")
+    private ThreadPoolTaskExecutor executor;
+
+    public DockerService(DockerRegistryProperties properties) {
+        DefaultDockerClientConfig.Builder builder = DefaultDockerClientConfig.createDefaultConfigBuilder()
+                .withDockerTlsVerify(false)
+                .withRegistryUrl(properties.getUrl())
+                .withRegistryPassword(properties.getPasword())
+                .withRegistryUsername(properties.getUsername());
+        Optional.ofNullable(host).ifPresent(builder::withDockerHost);
+        DefaultDockerClientConfig config = builder.build();
+
+        URI host = config.getDockerHost();
+        DockerHttpClient dockerHttpClient = new ApacheDockerHttpClient.Builder()
+                .dockerHost(host)
+                .sslConfig(config.getSSLConfig())
+                .maxConnections(50)
+                .build();
+        client = DockerClientImpl.getInstance(config, dockerHttpClient);
+        log.info("Initialized docker interface [{}]", host);
+    }
+
+    @Override
+    protected Overseer analyze(
+            AnalysisSyncRequestDTO analysis, File analysisDir, BiConsumer<String, String> sendCallback, Integer updateInterval, KrbConfig krbConfig
+    ) {
+        List<String> env = buildRuntimeEnvVariables(analysis.getDataSource(), krbConfig.getIsolatedRuntimeEnvs()).entrySet().stream().map(entry ->
+                entry.getKey() + "=" + entry.getValue()).collect(Collectors.toList()
+        );
+        Instant started = Instant.now();
+        Long id = analysis.getId();
+        String image = Optional.ofNullable(analysis.getDockerImage()).orElse(defaultImage);
+
+        StringBuffer stdout = new StringBuffer();
+        String script = WORKDIR + "/" + analysis.getExecutableFileName();
+        BiConsumer<String, String> callback = (stage, out) -> {
+            stdout.append(out);
+            sendCallback.accept(stage, out);
+        };
+
+        CompletableFuture<String> init = CompletableFuture.supplyAsync(
+                () -> {
+                    log.info("Execution [{}] use docker image [{}]", id, image);
```

Comment:
It would be technically incorrect to write "will use" or "uses", because at this point we have not yet established that image is successfully available. It is also not technically correct to say it will pull, because if image is already pulled and checksum matches, no pull will occur. So I opted for simpler language of the imperative mood.
Now that I have reflected on it a bit, I guess "to use" is a better wordking here.